armi/bookkeeping/db/database3.py

```diff
@@ -1264,11 +1264,18 @@ def _readParams(h5group, compTypeName, comps):
                 pDef = pDefs[paramName]
             except KeyError:
                 if re.match(r"^n[A-Z][a-z]?\d*", paramName):
```

Comment:
It seems that because we have the comps passed in we should also verify that the first one is a subclass of `Block` just to be safe. I'm not aware of other instances that this may come up in the future but this seems like a good trap for us to miss loading in number densities of some type later on without realizing where the error is occurring?

Reply:
Number densities are not ever read from these into the reactor state; they are only written to the database as a crutch for data visualization. Since we are only checking this condition if we have already found that there is no corresponding parameter definition, this seems perfectly fine to me.

Reply:
Oh I see. That makes sense, thanks!